app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -49,29 +44,21 @@
     private ListPreference langListPreference;
 
     @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
         ApplicationlessInjection
-                .getInstance(getActivity().getApplicationContext())
-                .getCommonsApplicationComponent()
-                .inject(this);
+            .getInstance(getActivity().getApplicationContext())
+            .getCommonsApplicationComponent()
+            .inject(this);
 
-        // Load the preferences from an XML resource
-        addPreferencesFromResource(R.xml.preferences);
+        // Set the preferences from an XML resource
+        setPreferencesFromResource(R.xml.preferences, rootKey);
 
-        themeListPreference = (ListPreference) findPreference(Prefs.KEY_THEME_VALUE);
+        themeListPreference = findPreference(Prefs.KEY_THEME_VALUE);
         prepareTheme();
 
-        //Check if the Author Name switch is enabled and appropriately handle the author name usage
-        SwitchPreference useAuthorName = (SwitchPreference) findPreference("useAuthorName");
-        EditTextPreference authorName = (EditTextPreference) findPreference("authorName");
-        authorName.setEnabled(defaultKvStore.getBoolean("useAuthorName", false));
-        useAuthorName.setOnPreferenceChangeListener((preference, newValue) -> {
-            authorName.setEnabled((Boolean)newValue);
-            return true;
-        });
+        SwitchPreference useAuthorName = findPreference("useAuthorName");
```

Comment:
This variables has an overly broad scope, I would even say you could inline (cursor on the variable name, `ctrl+alt+n` or right click>refactor>inline) it and the other variables you just disable when not logged in because their variable names are just repeating their Pref keys, the varaible name doesn't help describe them better so I think `findPreference("useAuthorName").setEnabled(false)` reads just as well

Reply:
Done!